packages/wallet-sdk/src/provider/MobileRelayUI.ts

```diff
@@ -0,0 +1,170 @@
+import { RedirectDialog } from "../components/RedirectDialog/RedirectDialog";
+import { ErrorHandler } from "../errors";
+import {
+  EthereumAddressFromSignedMessageRequest,
+  SignEthereumMessageRequest,
+  SignEthereumTransactionRequest,
+  SubmitEthereumTransactionRequest,
+} from "../relay/Web3Request";
+import {
+  EthereumAddressFromSignedMessageResponse,
+  SignEthereumMessageResponse,
+  SignEthereumTransactionResponse,
+  SubmitEthereumTransactionResponse,
+} from "../relay/Web3Response";
+import { AddressString, ProviderType } from "../types";
+import { WalletUI, WalletUIOptions } from "./WalletUI";
+
+// TODO: Implement & present in-page wallet picker instead of navigating to www.coinbase.com/connect-dapp
+export class MobileRelayUI implements WalletUI {
+  private readonly redirectDialog: RedirectDialog;
+  private attached = false;
+
+  constructor(_options: Readonly<WalletUIOptions>) {
+    this.redirectDialog = new RedirectDialog();
+  }
+
+  attach() {
+    if (this.attached) {
+      throw new Error("Coinbase Wallet SDK UI is already attached");
+    }
+    this.redirectDialog.attach();
+    this.attached = true;
+  }
+
+  private redirectToCoinbaseWallet(walletLinkUrl?: string): void {
+    const url = new URL("https://go.cb-w.com/walletlink");
+
+    url.searchParams.append("redirect_url", window.location.href);
+    if (walletLinkUrl) {
+      url.searchParams.append("wl_url", walletLinkUrl);
+    }
+
+    window.open(url.href, "_blank");
+  }
+
+  openCoinbaseWalletDeeplink(walletLinkUrl?: string): void {
+    this.redirectDialog.present({
+      title: "Redirecting to Coinbase Wallet...",
+      buttonText: "Open",
+      onButtonClick: () => {
+        this.redirectToCoinbaseWallet(walletLinkUrl);
+      },
+    });
+
+    setTimeout(() => {
+      this.redirectToCoinbaseWallet(walletLinkUrl);
+    }, 99);
```

Comment:
have encountered an issue where the browser seems to restrict calling window.open within the rxjs subscription block. 
per some SO posts, it's known that browser blocks web apps to call window.open from rxjs handlers.

Reply:
is the goal here to auto redirect if the user doesn't click the button?